tests/test_hcp.py

```diff
@@ -1,216 +1,213 @@
-    test_mount_bucket()
-    assert hcp_h.object_exists("a_sub_directory/a_file")
-    assert hcp_h.get_object("a_sub_directory/a_file")
-
-def test_get_folder_without_mounting() -> None:
-    _hcp_h = HCPHandler("credentials/testCredentials.json")
-    _without_mounting(_hcp_h.object_exists)
-    _without_mounting(_hcp_h.get_object)
-
-def test_get_file_in_sub_directory() -> None:
-    test_mount_bucket()
-    assert hcp_h.object_exists(test_file)
-    assert hcp_h.get_object(test_file)
-
-def test_download_file() -> None:
-    test_mount_bucket()
-    Path(result_path).mkdir()
-    hcp_h.download_file(test_file, result_path + test_file)
-    assert cmp(result_path + test_file, test_file_path)
-
-def test_download_file_without_mounting() -> None:
-    _hcp_h = HCPHandler("credentials/testCredentials.json")
-    _without_mounting(_hcp_h.download_file)
-
-def test_download_nonexistent_file() -> None:
-    test_mount_bucket()
+def test_get_file(custom_config : CustomConfig) -> None:
+    test_mount_bucket(custom_config)
+    assert custom_config.hcp_h.object_exists("a_sub_directory/a_file") 
+    assert custom_config.hcp_h.get_object("a_sub_directory/a_file") 
+
+def test_get_folder_without_mounting(custom_config : CustomConfig) -> None:
+    _hcp_h = custom_config.hcp_h 
+    _without_mounting(_hcp_h, HCPHandler.object_exists)
+    _without_mounting(_hcp_h, HCPHandler.get_object)
+
+def test_get_file_in_sub_directory(custom_config : CustomConfig) -> None:
+    test_mount_bucket(custom_config)
+    test_file = Path(custom_config.test_file_path).name 
+    assert custom_config.hcp_h.object_exists(test_file) 
+    assert custom_config.hcp_h.get_object(test_file) 
+
+def test_download_file(custom_config : CustomConfig) -> None:
+    test_mount_bucket(custom_config)
+    Path(custom_config.result_path).mkdir() 
+    test_file = Path(custom_config.test_file_path).name 
+    custom_config.hcp_h.download_file(test_file, custom_config.result_path + test_file) 
+    assert cmp(custom_config.result_path + test_file, custom_config.test_file_path) 
+
+def test_download_file_without_mounting(custom_config : CustomConfig) -> None:
+    _hcp_h = custom_config.hcp_h 
+    _without_mounting(_hcp_h, HCPHandler.download_file)
+
+def test_download_nonexistent_file(custom_config : CustomConfig) -> None:
+    test_mount_bucket(custom_config)
     try:
-        hcp_h.download_file("aFileThatDoesNotExist", result_path + "aFileThatDoesNotExist")
+        custom_config.hcp_h.download_file("aFileThatDoesNotExist", custom_config.result_path + "aFileThatDoesNotExist") 
     except:
         assert True
     else: # pragma: no cover
         assert False
 
-def test_download_folder() -> None:
-    test_mount_bucket()
-    hcp_h.download_folder("a folder of data/", result_path)
+def test_download_folder(custom_config : CustomConfig) -> None:
+    test_mount_bucket(custom_config)
+    custom_config.hcp_h.download_folder("a folder of data/", custom_config.result_path) 
```

Comment:
A lot of these tests lack asserts. Not necessary, but consider having them where its viable, even if its basic